drivers/modem/modem_context.c

```diff
@@ -20,23 +20,38 @@ LOG_MODULE_REGISTER(modem_context, CONFIG_MODEM_LOG_LEVEL);
 
 static struct modem_context *contexts[CONFIG_MODEM_CONTEXT_MAX_NUM];
 
-char *modem_context_sprint_ip_addr(const struct sockaddr *addr)
+int modem_context_sprint_ip_addr(const struct sockaddr *addr, char *buf, int buf_size)
 {
-	static char buf[NET_IPV6_ADDR_LEN];
+	char unknown_str[] = "unk";
 
 	if (addr->sa_family == AF_INET6) {
-		return net_addr_ntop(AF_INET6, &net_sin6(addr)->sin6_addr,
-				     buf, sizeof(buf));
+		if (buf_size < NET_IPV6_ADDR_LEN) {
+			return -ENOMEM;
+		}
+
+		net_addr_ntop(AF_INET6, &net_sin6(addr)->sin6_addr,
+				     buf, buf_size);
+		buf[buf_size - 1] = '\0';
```

Comment:
This should not be necessary; instead, it should check the return value of `net_addr_ntop` and return an error code upon failure in which case the content of `buf` should not be used.

Reply:
Seems reasonable I just overlooked this review comment, processed it